src/utils/log.h

```diff
@@ -30,21 +30,64 @@
 #ifndef G_UTILS_LOG_H
 #define G_UTILS_LOG_H
 
+#include <cstdio>
+#include <string>
+#include <type_traits>
+#include <utility>
 
-namespace giada {
-namespace u {
-namespace log 
-{
-/* init
-Initializes logger. Mode defines where to write the output: LOG_MODE_STDOUT,
-LOG_MODE_FILE and LOG_MODE_MUTE. */
+#include "utils/fs.h"
+#include "core/const.h"
 
-int init(int mode);
+namespace giada::u::log {
 
-void close();
+	inline FILE* f;
+	inline int   mode;
+	inline bool  stat;
 
-void print(const char* format, ...);
-}}}  // giada::u::log::
+	/* init
+	Initializes logger. Mode defines where to write the output: LOG_MODE_STDOUT,
+	LOG_MODE_FILE and LOG_MODE_MUTE. */
+	int init(int mode);
 
+	void close();
+
+	// Use forwarding references && to avoid useless string copy
+	static constexpr auto string_to_c_str = [] (auto&& s) {
+		// Remove any reference and const-ness, since the function can
+		// handle l-value and r-value, const or not.
+		// Use std::remove_cvref instead, when sitching to C++20...
+		if constexpr (std::is_same_v<std::remove_const_t<std::remove_reference_t<
+			decltype(s)>>,
+			std::string>)
```

Comment:
I have no idea how I can indent nicely this with tabs, to have for example `std::string` aligned with `std::remove_const_t`.